lib/lotus/entity.rb

```diff
@@ -144,7 +144,15 @@ def attributes(*attrs)
           self.attributes.merge attrs
 
           attrs.each do |attr|
-            attr_accessor(attr) if define_attribute?(attr)
+            if define_attribute?(attr)
+              attr_reader(attr)
+              self.class_eval %Q{
+                def #{attr}=(val)
+                  @initialized_attributes << __callee__
```

Comment:
@AlfonsoUceda where do you define `@initialized_attributes = Set.new`? It seems to me you need to define it advance just like you do with the constructor.